### ISSUE TITLE
Decisiones y modificaciones sobre la migración de la lista de operadores

### ISSUE THREAD

**Issue by Siotma** [asrob-uc3m/operadores#14]
Viene de https://github.com/asrob-uc3m/impresoras-asrob/issues/29#issuecomment-347841784

La [nueva página de operadores](http://asrob.uc3m.es/printers/operators.html) está activa y funcionando.

En este momento, nos enfrentamos a un proceso de transición en el que se abren distintas tareas:

- [ ] Si se decide hacer la transición, habría que indicar con una nota en la antigua página que ya no se actualizará más, y poner un link a la nueva.
- [ ] Si se decide hacer la transición, habría que establecer una fecha límite para que los operadores que figuran en la actual lista migren automáticamente a la nueva al estilo de https://github.com/asrob-uc3m/operadores/issues/4
- [ ] Asimismo, habría que establecer la duración máxima del título de operador entre renovaciones.
- [ ] Por último, habría que detallar, el procedimiento de formación, tanto para el nuevo operador como para el formador, e indicarlo claramente en la web (#11) y en el repo de [operadores](https://github.com/asrob-uc3m/operadores/)

Ideas?

**Comment by David-Estevez** [comment]
> Si se decide hacer la transición, habría que establecer una fecha límite para que los operadores que figuran en la actual lista migren automáticamente a la nueva al estilo de #4

Mucha de la gente que aparece  en la lista se formó con las Makerbot / hace mucho que no pasa por la asociación por lo que, en mi opinión, toda esa gente debería hacer un curso de reciclaje si quiere seguir siendo operador, siguiendo el nuevo procedimiento.

**Comment by Siotma** [comment]
> toda esa gente debería hacer un curso de reciclaje si quiere seguir siendo operador, siguiendo el nuevo procedimiento.
Estoy de acuerdo con eso. Sin embargo, dado que es nuestro actual sistema de gestión de operadores, no podríamos obligar a unos sí y a otros no. Por ello, se me ocurre que el plazo que demos se aplique a todo aquel que figure en la antigua lista. Workflow:

1. Antes de la fecha límite, cualquiera de la lista puede renovar el título de operador simplemente con pedirlo al estilo de #4 
2. El día de la fecha límite, aquellos renovados según el punto 1 deben haberse reciclado según el nuevo procedimiento (ello no sería un problema dado que todos utilizamos asiduamente las impresoras).
3. Más allá de la fecha límite, cualquiera que quiera renovar deberá reciclarse.

**Comment by David-Estevez** [comment]
Yo en ese aspecto no estoy del todo de acuerdo. No creo que se ha hecho ninguna excepción ya que las personas a las que se ha hecho operadoras "directamente", y que ya están incluídas, han recibido su formación y/o usado la impresora recientemente. El resto de personas hace **años** que no usan una impresora de ASROB (reitero lo de que muchos se formaron con Makerbots y Replicator G) y creo que sería bastante peligroso para las impresoras hacerles operadores directamente como al resto.

**Comment by Siotma** [comment]
De acuerdo, entonces podría quedar como: aquel que quiera migrar de la antigua lista a la nueva debe poder acreditar que ha utilizado las impresoras recientemente.

**Comment by David-Estevez** [comment]
Eso me parece mucho mejor para la integridad de las impresoras, sí :+1:

**Comment by Siotma** [comment]
> Asimismo, habría que establecer la duración máxima del título de operador entre renovaciones.

Según el código de generación de la página, actualmente esta duración es de un año. Sin embargo, creo que deberíamos consensuarlo para hacerlo más robusto. Me refiero a que en el momento en que el software de uso de las impresoras cambie significativamente, la renovación debería hacerse extensiva a todo el mundo para evitar cosas como [la que mencionabas](https://github.com/asrob-uc3m/operadores/issues/14#issuecomment-348161099).

Se me ha ocurrido hacer un sistema de etiquetas con versiones (ej v1, ej Repetier1.6, etc) que indiquen en qué versión del software se ha formado cada uno. Una vez implantado algo así, se podría comprobar automáticamente para la generación de la página. Por tanto, en el momento en que la versión más alta cambie, todos los operadores pasarían a ser antiguos y @asrobuc3m debería acreditar al menos a uno de ellos (se entiende que alguno de ellos llevaría a cabo el proceso de actualización de softwares) y por tanto una vez acreditado se podría continuar con el procedimiento habitual de formaciones y renovaciones.

**Comment by David-Estevez** [comment]
Me gusta, el modelo de etiquetas sería extensible también a modelos de impresora, por ejemplo.

La única pega que le veo es que convierta el software de generación de la página en un infierno de condiciones (que si X fecha, que si Y formó a Z, que si etiquetas de versiones...)

**Comment by Siotma** [comment]
Sobre modelos de impresora, al menos a nivel de usuario, hay poca diferencia, pero claro igual de cara a futuro también habría que incluir esa distinción 😅 

Habría que probar a ver hasta dónde somos capaces de llegar antes de enloquecer 🤣

**Comment by Siotma** [comment]
Responsabilidades del formador v0.1: 
- Quedar con el padawan un día o dos (los que hagan falta), enseñarle a utilizar las impresoras basándose en el [manual de usuario](https://asrob-uc3m.gitbooks.io/impresoras-user-manual-i3/content/) y finalmente asegurarse de que ha aprendido pidiéndole que imprima él solo bajo su supervisión.
- Durante el proceso, la comunicación entre padawan y formador se hará mediante la issue de formación; el formador documentará con comentarios el proceso de formación (hoy hemos visto tal cosa , hemos impreso tal pieza ...) y alguno de los dos subirá a la issue fotos de las piezas que se impriman durante la formación.
- Enseñarle no solo el manejo de la impresora sino también precauciones de protección personal (partes calientes, no pillarse los dedos con los ejes); protección de la impresora (buen estado de los ventiladores, buen estado de los finales de carrera, uso del botón parada de emergencia); protección del entorno (no dejar la impresora desatendida), etc...
- Enseñarle buenas prácticas de impresión (posicionado de las piezas, soportes, densidad de relleno...) y fuentes donde obtener diseños 3D (thingiverse y similares).

Se me ha ocurrido así a bote pronto. Creéis que sería buena idea redactar una guía para formadores, o sería suficiente con incluir estas pautas en el manual de usuario?

**Comment by Siotma** [comment]
A estas alturas, y si haber encontrado oposición, he editado la [página de Operadores de la Wiki](http://asrob.uc3m.es/index.php/Operadores) y la de [Antiguos Operadores](http://asrob.uc3m.es/index.php/Antiguos_Operadores) para indicar que esas páginas ya no se utilizarán, y se usará [la nueva](http://asrob.uc3m.es/printers/operators.html), enlazándola correspondientemente.

En relación a esto, la [antigua página de operadores](http://asrob.uc3m.es/index.php/Operadores), tenía dos campos especiales: el de responsable y el de operadores honoríficos. ¿Se debería portar esta información a la nueva página antes de que se pierda definitivamente?

Viendo que os ha gustado el anterior mensaje, y enlazando con https://github.com/asrob-uc3m/operadores/issues/11#issuecomment-347957550, creo que la documentación del proceso de formación se merece una issue exclusiva para ello: #22

Finalmente, quedaría antes de cerrar esta issue, discutir la duración del título de operador. Actualmente esta duración es por tiempo absoluto: 1 año. Creo que hasta que surja la necesidad de ajustar este criterio, el uso del tiempo puede ser un mínimo producto viable. En caso de demostrarse que con ello es suficiente, se podría evitar aumentar la complejidad del sistema de forma innecesaria.

Duda: una vez que esta issue se cierre, las decisiones como: 1. método de transición de la antigua lista a esta y 2. duración del título de operador, deberían transcribirse a un soporte más duradero, tal vez los [estatutos](https://github.com/asrob-uc3m/estatutos)?
